boot/core/src/boot/core.clj

```diff
@@ -1218,10 +1218,12 @@
   "A file filtering function factory. FIXME: more documenting here."
   [mkpred]
   (fn [criteria files & [negate?]]
-    (let [tmp?   (partial satisfies? tmpd/ITmpFile)
-          ->file #(if (tmp? %) (io/file (tmp-path %)) (io/file %))
-          pred (apply juxt (mapv mkpred criteria))]
-      (filter-files #(pred (->file %)) files negate?))))
+    (if (empty? criteria)
```

Comment:
Would that not result in a `nil` return if the seq was empty? I'm still new to Clojure and Boot, so I may be missing something.